# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: media/mtransport/ipc/WebrtcTCPSocketParent.cpp
# Commit: da41f0fab427
# Full Hash: da41f0fab427b199f1a0c6156467a0ad9a9e6891
# Author: Byron Campen [:bwc] <docfaraday@gmail.com>
# Date: 2019-09-18 21:49:27
# Regressor Bug: 1569183
# File Overlap Count: 2
# Description:
#   Bug 1569183: Make proxy config parameters optional in WebrtcTCPSocket to get the NAT simulator working again. r=mjf,mayhemer
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D45104
# ==============================================================================

diff -r b83124cf3a89 -r da41f0fab427 media/mtransport/ipc/WebrtcTCPSocketParent.cpp
--- a/media/mtransport/ipc/WebrtcTCPSocketParent.cpp	Tue Sep 17 18:17:24 2019 +0000
+++ b/media/mtransport/ipc/WebrtcTCPSocketParent.cpp	Tue Sep 17 18:17:37 2019 +0000
@@ -20,15 +20,19 @@
 mozilla::ipc::IPCResult WebrtcTCPSocketParent::RecvAsyncOpen(
     const nsCString& aHost, const int& aPort, const nsCString& aLocalAddress,
     const int& aLocalPort, const bool& aUseTls,
-    const LoadInfoArgs& aLoadInfoArgs, const nsCString& aAlpn,
-    const int& aProxyPolicy) {
+    const Maybe<LoadInfoArgs>& aLoadInfoArgs, const Maybe<nsCString>& aAlpn,
+    const Maybe<int>& aProxyPolicy) {
   LOG(("WebrtcTCPSocketParent::RecvAsyncOpen %p to %s:%d\n", this, aHost.get(),
        aPort));
 
   MOZ_ASSERT(mChannel, "webrtc TCP socket should be non-null");
+  Maybe<NrSocketProxyConfig::ProxyPolicy> proxyPolicy;
+  if (aProxyPolicy.isSome()) {
+    proxyPolicy =
+        Some(static_cast<NrSocketProxyConfig::ProxyPolicy>(*aProxyPolicy));
+  }
   mChannel->Open(aHost, aPort, aLocalAddress, aLocalPort, aUseTls,
-                 aLoadInfoArgs, aAlpn,
-                 static_cast<NrSocketProxyConfig::ProxyPolicy>(aProxyPolicy));
+                 aLoadInfoArgs, aAlpn, proxyPolicy);
 
   return IPC_OK();
 }
@@ -65,13 +69,15 @@
   CleanupChannel();
 }
 
-WebrtcTCPSocketParent::WebrtcTCPSocketParent(dom::TabId aTabId) {
+WebrtcTCPSocketParent::WebrtcTCPSocketParent(const Maybe<dom::TabId>& aTabId) {
   MOZ_COUNT_CTOR(WebrtcTCPSocketParent);
 
   LOG(("WebrtcTCPSocketParent::WebrtcTCPSocketParent %p\n", this));
 
   mChannel = new WebrtcTCPSocket(this);
-  mChannel->SetTabId(aTabId);
+  if (aTabId.isSome()) {
+    mChannel->SetTabId(*aTabId);
+  }
 }
 
 WebrtcTCPSocketParent::~WebrtcTCPSocketParent() {